### PR TITLE
Add a session filter for archives

### PR DESCRIPTION
**Pull request by Lucashuang0802**

### CODE REVIEW

File: lib/opentok/client.rb [hunk 1]
```diff
@@ -102,10 +102,11 @@ def get_archive(archive_id)
       raise OpenTokError, "Failed to connect to OpenTok. Response code: #{e.message}"
     end
 
-    def list_archives(offset, count)
+    def list_archives(offset, count, session_id)
       query = Hash.new
       query[:offset] = offset unless offset.nil?
       query[:count] = count unless count.nil?
+      query[:session_id] = session_id unless session_id.nil?
```
Comment (aiham) [inline lib/opentok/client.rb]: @Lucashuang0802 Shouldn't this be `query[:sessionId]`?

File: lib/opentok/archives.rb [hunk 2]
```diff
@@ -96,11 +96,13 @@ def find(archive_id)
     #   the most recent archive. If you do not specify an offset, 0 is used.
     # @option options [integer] :count Optional. The number of archives to be returned. The maximum
     #   number of archives returned is 1000.
+    # @option options [String] :session_id Optional. The session ID that archives belong to. This is
+    #   useful when listing multiple archives for a {https://tokbox.com/developer/guides/archiving/#automatic-archives automatically archived session}
```
Comment (aiham) [inline lib/opentok/archives.rb]: Should be `for an`.

File: lib/opentok/archives.rb [hunk 3]
```diff
@@ -96,11 +96,13 @@ def find(archive_id)
...
     #
     # @return [ArchiveList] An ArchiveList object, which is an array of Archive objects.
     def all(options = {})
-      raise ArgumentError, "Limit is invalid" unless options[:count].nil? or (0..100).include? options[:count]
-      archive_list_json = @client.list_archives(options[:offset], options[:count])
+      raise ArgumentError, "Limit is invalid" unless options[:count].nil? or (0..1000).include? options[:count]
```
Comment (aiham) [inline lib/opentok/archives.rb]: Why have you increased it to 1000?

File: spec/cassettes/OpenTok_Archives/when_many_archives_are_created/should_return_session_archives.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      - application/json
+      Transfer-Encoding:
+      - chunked
+      Connection:
+      - keep-alive
+    body:
+      encoding: UTF-8
+      string: |-
+        {
+          "count" : 6,
```
Comment (aiham) [inline spec/cassettes/OpenTok_Archives/when_many_archives_are_created/should_return_session_archives.yml]: Shouldn't this be 3?

File: spec/opentok/archives_spec.rb [hunk 5]
```diff
@@ -124,6 +124,12 @@
       expect(archive_list).to be_an_instance_of OpenTok::ArchiveList
       expect(archive_list.count).to eq 4
     end
+
+    it "should return session archives", :vcr => { :erb => { :version => OpenTok::VERSION } } do
+      archive_list = archives.all :session_id => session_id
+      expect(archive_list).to be_an_instance_of OpenTok::ArchiveList
+      expect(archive_list.total).to eq 6
```
Comment (aiham) [inline spec/opentok/archives_spec.rb]: Shouldn't this be 3?

**Comment by Lucashuang0802** [comment]
Forget to mention @MrWilhelm as this implementation is based on #130. Thanks for starting it! 💯 😄 🎉

File: lib/opentok/archives.rb (hunk 3 above)
Comment (Lucashuang0802) [inline lib/opentok/archives.rb]: If I understand correctly, the total number of archives returned could be up to 1000.

File: lib/opentok/client.rb (hunk 1 above)
Comment (Lucashuang0802) [inline lib/opentok/client.rb]: Just double check with the docs, it is indeed `sessionId`

File: lib/opentok/archives.rb (hunk 2 above)
Comment (Lucashuang0802) [inline lib/opentok/archives.rb]: 👍

File: spec/cassettes/OpenTok_Archives/when_many_archives_are_created/should_return_session_archives.yml (hunk 4 above)
Comment (Lucashuang0802) [inline spec/cassettes/OpenTok_Archives/when_many_archives_are_created/should_return_session_archives.yml]: Yes, I found the number abnormal but the test passes. So I did not pay attention to that. It turns out that test cases did not actual cover that. The change has pushed. 😃

**Comment by Lucashuang0802** [comment]
#126

**Comment by aiham** [comment]
Thanks @MrWilhelm and @kouno!

**Pull request merged**